### PR TITLE
Report number of people who posted for the first time in this Slack

### PR DESCRIPTION
**Pull request by royrapoport**

### CODE REVIEW

File: firstpost.py [hunk 1]
```diff
@@ -102,6 +102,22 @@ def message(self, message):
     def get_channel(self, cid):
         return self.get(cid)
 
+    def firstpost_count(self, start_date, days):
+        """
+        based on a yyyy-mm-dd start_date and an int days, report how many
+        people had their first post in this period
+        """
+        (y, m, d) = [int(x) for x in start_date.split("-")]
```
Comment (jdq22) [inline firstpost.py:110]: This is really cool (and pythonic)!

File: firstpost.py [hunk 2]
```diff
@@ -102,6 +102,22 @@ def message(self, message):
...
+        dt = datetime.datetime(y, m, d, 0, 0, 0)
+        report_start_ts = dt.timestamp()
+        report_end_ts = report_start_ts + 86400 * days
+        matches =[]
+        for item in self.ddb.items(self.ddb.get_table()):
```
Comment (jdq22) [inline firstpost.py:115]: Can you remind me why DDB can't do this query?

File: report.py [hunk 3]
```diff
@@ -131,6 +131,9 @@ def set_start_date(self, start_date):
     def set_end_date(self, end_date):
         self._data['end_date'] = end_date
 
+    def set_days(self, days):
```
Comment (jdq22) [inline report.py:134]: What is this for?

File: firstpost.py (hunk 2 above)
Comment (royrapoport) [reply firstpost.py:115]: DDB is general-purpose, and this query is kind of more specific than that.

File: report.py (hunk 3 above)
Comment (royrapoport) [reply report.py:134]: I wanted the report json to include the number of days for the report (we have the start_date and end_date and could calculate the number of days from that, but this seemed easier and cleaner)

Comment (jdq22) [reply report.py:134]: Where is `self._data['days']` being used?

Comment (royrapoport) [reply report.py:134]: See the slack_global_report.py commit

**Review by jdq22 — APPROVED** [review]

**Pull request merged**